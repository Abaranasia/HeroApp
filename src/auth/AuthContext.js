import { createContext } from "react";

export const AuthContext= createContext();
// lleva may√∫suclas porque es un componente, en este caso un high order component (apertura y cierre)