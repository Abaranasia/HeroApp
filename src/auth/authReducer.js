import { types } from "../types/types";

/** Reducer para tener bajo los estados de autenticación en la app
 * Toma los estados (login y logout) desde types
 * Este reducer estará compartido en toda la aplicación gracias a un context (AuthContext)
*/

export const authReducer = (state= {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false
            }
        default:
            return state;
    }
}