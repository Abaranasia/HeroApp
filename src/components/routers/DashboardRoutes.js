import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';

import { Navbar } from '../ui/Navbar';
import { DcScreen } from '../dc/DcScreen';
import { HeroScreen } from '../heroes/HeroScreen';
import { MarvelScreen } from '../marvel/MarvelScreen';
import { SearchScreen } from '../search/SearchScreen';


export const DashboardRoutes = () => {
    return (
        <>
          <Navbar/> 
          <div className="container mt-3">
              <Switch>
                  <Route exact path="/marvel" component= { MarvelScreen } />
                  <Route exact path="/dc" component= { DcScreen } />
                  <Route exact path="/hero/:heroId" component= { HeroScreen } />
                  <Route exact path="/search" component= { SearchScreen } />
                  
                  <Redirect to="/marvel" />
              </Switch>
          </div> 
        </>
    )
}
