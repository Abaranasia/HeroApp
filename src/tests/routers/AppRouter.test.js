import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../components/routers/AppRouter";


describe('Pruebas en AppRouter', () => {
 
    test('Debe mostrar loginScreen si no estoy autenticado ', () => {

        const contextValue = { // valores necesarios para enviar en el context provider
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        }

        /** Este componente requiere de una provisi칩n por parte de AuthContext, por lo que
         *  habr치 que importar el componente e incluirlo en el shallow. Dado que es un
         *  High Order Component, shallow no nos servir치, por lo que debemos usar mount()
         */

        const wrapper = mount( 
            <AuthContext.Provider value = { contextValue }>
                <AppRouter/> 
            </AuthContext.Provider>
        );
        
        //console.log(wrapper.html())
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h2').exists()).toBe(true);
        expect(wrapper.find('h2').text()).toBe('Login')   
    });

    test('Debe mostrar el componente de Marvel si est치 autenticado', () => {

        const contextValue = { // valores necesarios para enviar en el context provider
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Ran'
            }
        }

        const wrapper = mount( 
            <AuthContext.Provider value = { contextValue }>
                <AppRouter/> 
            </AuthContext.Provider>
        );
        
       // console.log(wrapper.html())
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true); // suficiente saber que se ha cargado el navbar
    });

});