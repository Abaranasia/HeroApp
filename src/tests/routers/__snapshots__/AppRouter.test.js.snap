// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe mostrar el componente de Marvel si está autenticado 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2"
      >
        <ul
          className="navbar-nav ms-auto"
        >
          <span
            className="nav-item nav-link text-info"
          >
            Ran
          </span>
          <button
            className="nav-item nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>,
    <div
      className="container mt-3"
    >
      <div>
        <h2>
          Héroes de Marvel Cómics
        </h2>
        <div
          className="row row-cols-1 row-cols-md-2 g-4 animate__animated animate__fadeIn"
        >
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  <img
                    alt="Spider Man"
                    className="card-img-top"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Peter Parker
                     
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  <img
                    alt="Captain America"
                    className="card-img-top"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Steve Rogers
                     
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  <img
                    alt="Iron Man"
                    className="card-img-top"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Tony Stark
                     
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  <img
                    alt="Thor"
                    className="card-img-top"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Thor Odinson
                     
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  <img
                    alt="Hulk"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Bruce Banner
                     
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  <img
                    alt="Wolverine"
                    className="card-img-top"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    James Howlett
                     
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  <img
                    alt="Daredevil"
                    className="card-img-top"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Matthew Michael Murdock
                     
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  <img
                    alt="Hawkeye"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Clinton Francis Barton
                     
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  <img
                    alt="Cyclops"
                    className="card-img-top"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Scott Summers
                     
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-2"
            style={
              Object {
                "maxWidth": 250,
              }
            }
          >
            <div
              className="row no-gutters"
            >
              <div
                className=""
              >
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  <img
                    alt="Silver Surfer"
                    className="card-img-top"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </a>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Norrin Radd
                     
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;

exports[`Pruebas en AppRouter Debe mostrar loginScreen si no estoy autenticado  1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h2>
      Login
    </h2>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
